kore/src/Kore/Step/Axiom/Registry.hs

```diff
@@ -128,46 +132,74 @@ axiomToIdAxiomPatternPair axiom =
+        functionRules =
+            sortOn getPriorityOfRule
+            . filter (not . ignoreDefinition)
+            $ unProcessedFunctionRules
+
+-- | Converts a registry of processed 'EqualityRule's to one of
+-- 'BuiltinAndAxiomSimplifier's
+processedAxiomsToEvaluators
+    :: Map.Map AxiomIdentifier PartitionedEqualityRules
     -> Map.Map AxiomIdentifier BuiltinAndAxiomSimplifier
-axiomPatternsToEvaluators =
-    Map.fromAscList . mapMaybe equalitiesToEvaluators . Map.toAscList
+processedAxiomsToEvaluators =
+    mapMaybe equalitiesToEvaluators
   where
     equalitiesToEvaluators
-        :: (AxiomIdentifier, [EqualityRule Variable])
-        -> Maybe (AxiomIdentifier, BuiltinAndAxiomSimplifier)
+        :: PartitionedEqualityRules
+        -> Maybe BuiltinAndAxiomSimplifier
     equalitiesToEvaluators
-        (symbolId, filter (not . ignoreEqualityRule) -> equalities)
+        PartitionedEqualityRules { functionRules, simplificationRules }
       =
         case (simplificationEvaluator, definitionEvaluator) of
             (Nothing, Nothing) -> Nothing
-            (Just evaluator, Nothing) -> Just (symbolId, evaluator)
-            (Nothing, Just evaluator) -> Just (symbolId, evaluator)
+            (Just evaluator, Nothing) -> Just evaluator
+            (Nothing, Just evaluator) -> Just evaluator
             (Just sEvaluator, Just dEvaluator) ->
-                Just (symbolId, simplifierWithFallback sEvaluator dEvaluator)
+                Just (simplifierWithFallback sEvaluator dEvaluator)
       where
-        simplifications, evaluations :: [EqualityRule Variable]
-        (simplifications, filter (not . ignoreDefinition) -> evaluations) =
-            partition isSimplificationRule equalities
-          where
-            isSimplificationRule (EqualityRule EqualityPattern { attributes }) =
-                isSimplification
-              where
-                Simplification { isSimplification } =
-                    Attribute.simplification attributes
-        simplification :: [BuiltinAndAxiomSimplifier]
-        simplification = simplificationEvaluation <$> simplifications
         simplificationEvaluator =
-            if null simplification
+            if null simplificationRules
                 then Nothing
-                else Just (firstFullEvaluation simplification)
+                else
+                    Just . firstFullEvaluation
+                    $ simplificationEvaluation
+                    <$> simplificationRules
         definitionEvaluator =
-            if null evaluations
+            if null functionRules
                 then Nothing
-                else Just (definitionEvaluation evaluations)
+                else Just $ definitionEvaluation functionRules
+
+axiomPatternsToEvaluators
+    :: Map.Map AxiomIdentifier [EqualityRule Variable]
+    -> Map.Map AxiomIdentifier BuiltinAndAxiomSimplifier
+axiomPatternsToEvaluators =
+    processedAxiomsToEvaluators . processAxiomPatterns
```

Comment:
I think I would prefer this:

```suggestion
    mapMaybe (processedAxiomsToEvaluators . processAxiomPatterns)
```

with the types of `processedAxiomsToEvaluators` and `processAxiomPatterns` changed accordingly, because those two functions don't actually do anything with the `Map`.